src/lib/Persistence/Legacy/Content/Mapper/ResolveVirtualFieldSubscriber.php

```diff
@@ -0,0 +1,222 @@
+            );
+
+            if ($result === true) {
+                $storageValue = new StorageFieldValue();
+                $converter = $this->converterRegistry->getConverter($fieldDefinition->fieldType);
+                $converter->toStorageValue(
+                    $field->value,
+                    $storageValue
+                );
+
+                $this->contentGateway->updateField(
+                    $field,
+                    $storageValue
+                );
+            }
+        }
+
+        $storage->getFieldData(
+            $content->versionInfo,
+            $field,
+            []
+        );
+
+        $event->setField($field);
+    }
+
+    public function resolveVirtualExternalStorageField(ResolveMissingFieldEvent $event): void
+    {
+        $field = $event->getField();
+
+        if ($field === null) {
+            // Nothing to resolve
+            return;
+        }
+
+        if ($field->id !== null) {
+            // Not a virtual field
+            return;
+        }
+
+        $fieldDefinition = $event->getFieldDefinition();
+        $storage = $this->storageRegistry->getStorage($fieldDefinition->fieldType);
+
+        if ($storage instanceof NullStorage) {
+            // Not an external storage
+            return;
+        }
+
+        if (!$storage instanceof DefaultDataFieldStorage) {
+            return;
+        }
+
+        $content = $event->getContent();
+
+        $storage->getDefaultFieldData(
+            $content->versionInfo,
+            $field
+        );
+
+        $event->setField($field);
+
+        // Do not persist the external storage field
+        $event->stopPropagation();
+    }
+
+    /**
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\NotFoundException
+     */
+    private function createEmptyField(
+        VersionInfo $versionInfo,
+        FieldDefinition $fieldDefinition,
+        string $languageCode
+    ): Field {
+        $field = new Field();
+        $field->fieldDefinitionId = $fieldDefinition->id;
+        $field->type = $fieldDefinition->fieldType;
+        $field->value = $this->getDefaultValue($fieldDefinition);
+        $field->languageCode = $languageCode;
+        $field->versionNo = $versionInfo->versionNo;
+
+        return $field;
+    }
+
+    /**
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\NotFoundException
+     */
+    private function getDefaultValue(FieldDefinition $fieldDefinition): FieldValue
+    {
+        $value = clone $fieldDefinition->defaultValue;
+        $storageValue = $this->getDefaultStorageValue();
+
+        $converter = $this->converterRegistry->getConverter($fieldDefinition->fieldType);
+        $converter->toStorageValue($value, $storageValue);
+        $converter->toFieldValue($storageValue, $value);
+
+        return $value;
+    }
+
+    private function getDefaultStorageValue(): StorageFieldValue
+    {
+        $storageValue = new StorageFieldValue();
+        $storageValue->dataFloat = 0;
```

Comment:
In https://github.com/ezsystems/ezplatform-kernel/pull/397/files#diff-a6c919d22de3f556e4e01343e3b9e5453fee9cee0c74383447605026b6941fc0R207 we had a null. Wouldn't make sense to have float value due to strict typing here?
```suggestion
        $storageValue->dataFloat = 0.0;
```